tests/integration/targets/ec2_eip/tasks/update_reverse_dns_record.yml

```diff
@@ -0,0 +1,43 @@
+- name: Test EIP allocation
+  block:
+    # ------------------------------------------------------------------------------------------
+    # Allocate EIP with reverse DNS record
+    # ------------------------------------------------------------------------------------------
+    - name: Allocate a new EIP and modify it's reverse DNS record - check_mode
+      amazon.aws.ec2_eip:
+        state: present
+        domain_name: "{{ test_domain }}"
+        tags: "{{ eip_test_tags }}"
+      register: eip
+      check_mode: true
+
+    - ansible.builtin.assert:
+        that:
+          - eip is changed
+
+    - name: Ensure no new EIP was created
+      ansible.builtin.include_tasks: tasks/common.yml
+      vars:
+        has_no_new_eip: true
+
+    - name: Allocate a new EIP and modify it's reverse DNS record
```

Comment:
Can you also please add an idempotency check?

Reply:
moved to slack channel for discussion

Reply:
@alinabuzachis I have updated the tested to include more testing. The tests pass locally
```
        "ec2:DescribeVpcs",
        "ec2:DeleteVpc"
    ],
    "vpc": {}
}

PLAY RECAP *********************************************************************
testhost                   : ok=45   changed=17   unreachable=0    failed=0    skipped=16   rescued=0    ignored=0   

AWS ACTIONS: ['ec2:AllocateAddress', 'ec2:AttachInternetGateway', 'ec2:AuthorizeSecurityGroupIngress', 'ec2:CreateInternetGateway', 'ec2:CreateNetworkInterface', 'ec2:CreateSecurityGroup', 'ec2:CreateSubnet', 'ec2:CreateVpc', 'ec2:DeleteInternetGateway', 'ec2:DeleteNetworkInterface', 'ec2:DeleteSecurityGroup', 'ec2:DeleteSubnet', 'ec2:DeleteVpc', 'ec2:DescribeAddresses', 'ec2:DescribeAddressesAttribute', 'ec2:DescribeAvailabilityZones', 'ec2:DescribeInternetGateways', 'ec2:DescribeNetworkInterfaces', 'ec2:DescribeSecurityGroups', 'ec2:DescribeSubnets', 'ec2:DescribeTags', 'ec2:DescribeVpcAttribute', 'ec2:DescribeVpcs', 'ec2:DetachInternetGateway', 'ec2:ModifyAddressAttribute', 'ec2:ModifyVpcAttribute', 'ec2:ReleaseAddress', 'ec2:ResetAddressAttribute', 'route53:ChangeResourceRecordSets', 'route53:GetChange', 'route53:ListHostedZones', 'route53:ListResourceRecordSets', 'sts:GetCallerIdentity']
Command exited with status 0 after 435.9944279193878 seconds.
```
But, for now, I have disabled the tests for reverse DNS record feature as it requires use of registered domain and a corresponding hosted zone to successfully add `reverse DNS record` to an EIP.
Slack thread started for discussion on how to handle this.